modules/tinymce/src/core/main/ts/textpatterns/core/InlinePattern.ts

```diff
@@ -49,7 +49,10 @@ const findPatternStartFromSpot = (dom: DOMUtils, pattern: InlinePattern, block:
   const startPattern = pattern.start;
   const startSpot = TextSearch.repeatLeft(dom, spot.container, spot.offset, matchesPattern(startPattern), block);
   return startSpot.bind((spot) => {
-    if (spot.offset >= startPattern.length) {
+    const preStartLength = block.textContent?.indexOf(startPattern) ?? 0;
```

Comment:
I actually did not think about that, typing in the editor works as expected since after `foo` the user press `space` and it is parsed so entering the second part, works.

I tried pasting the entire string but it works as in the fiddle atm (only `bar` get parsed after I press space).

I wrote this test:
```
it('TINY-8949: the patterns should be applied to all the elements', () => {
    const editor = hook.editor();
    editor.options.set('text_patterns', [
      { start: '**', end: '**', format: 'bold' },
      { start: '***', end: '***', format: 'italic' },
    ]);

    Utils.setContentAndPressSpace(editor, 'a **test1** **test2**');
    TinyAssertions.assertContent(editor, '<p>a <strong>test1</strong> <strong>test2</strong>&nbsp;</p>');

    Utils.setContentAndPressSpace(editor, 'a ***test1*** ***test2***');
    TinyAssertions.assertContent(editor, '<p>a <em>test1</em> <em>test2</em>&nbsp;</p>');

    editor.options.unset('text_patterns');
  });
  ```
  but it seems to fail as it fails in `develop` (only `test2` get the style).
  
  is that the correct behavior?